recipes-multimedia/alsa/alsa-lib/0001-WIP-add-confs-for-gx-sound-card-and-axg-sound-card.patch

```diff
@@ -0,0 +1,201 @@
+From 34dbf4272c3036a7e8b208ad8dc35a035918af66 Mon Sep 17 00:00:00 2001
+From: chewitt <christianshewitt@gmail.com>
+Date: Thu, 14 May 2020 13:31:34 +0000
+Subject: [PATCH 1/3] WIP: add confs for gx-sound-card and axg-sound-card
+
+---
+ src/conf/cards/axg-sound-card.conf | 87 ++++++++++++++++++++++++++++++
+ src/conf/cards/gx-sound-card.conf  | 87 ++++++++++++++++++++++++++++++
+ 2 files changed, 174 insertions(+)
+ create mode 100644 src/conf/cards/axg-sound-card.conf
+ create mode 100644 src/conf/cards/gx-sound-card.conf
+
+diff --git a/src/conf/cards/axg-sound-card.conf b/src/conf/cards/axg-sound-card.conf
+new file mode 100644
+index 00000000..f3caaed0
+--- /dev/null
++++ b/src/conf/cards/axg-sound-card.conf
+@@ -0,0 +1,87 @@
++# SPDX-License-Identifier: GPL-2.0
++# Copyright (C) 2020-present Team LibreELEC (https://libreelec.tv)
++
++#
++# Configuration for Amlogic AXG audio
++#
++
++axg-sound-card.pcm.default {
++	@args [ CARD ]
++	@args.CARD { type string }
++	type softvol
++	slave.pcm {
++		type plug
++		slave {
++			pcm {
++				type hw
++				card $CARD
++				device 0
++			}
++		}
++	}
++	control {
++		name "PCM Playback Volume"
++		card $CARD
++	}
++}
++
++<confdir:pcm/hdmi.conf>
++
++axg-sound-card.pcm.hdmi.0 {
++	@args [ CARD AES0 AES1 AES2 AES3 ]
++	@args.CARD { type string }
++	@args.AES0 { type integer }
++	@args.AES1 { type integer }
++	@args.AES2 { type integer }
++	@args.AES3 { type integer }
++	type hooks
++	slave.pcm {
++		type hw
++		card $CARD
++		device 0
++	}
++	hooks.0 {
++		type ctl_elems
++		hook_args [
++			{
++				name "IEC958 Playback Default"
++				interface PCM
++				lock true
++				preserve true
++				optional true
++				value [ $AES0 $AES1 $AES2 $AES3 ]
++			}
++		]
++	}
++}
++
++<confdir:pcm/iec958.conf>
++
++axg-sound-card.pcm.iec958.0 {
++	@args [ CARD AES0 AES1 AES2 AES3 ]
++	@args.CARD { type string }
++	@args.AES0 { type integer }
++	@args.AES1 { type integer }
++	@args.AES2 { type integer }
++	@args.AES3 { type integer }
++	type hooks
++	slave.pcm {
++		type hw
++		card $CARD
++		device 1
++	}
++	hooks.0 {
++		type ctl_elems
++		hook_args [
++			{
++				name "IEC958 Playback Default"
++				interface PCM
++				lock true
++				preserve true
++				optional true
++				value [ $AES0 $AES1 $AES2 $AES3 ]
++			}
++		]
++	}
++	hint.device 1
++}
```

Comment:
```
[  830.238251][  T994]  fe.dai-link-1: ASoC: no backend DAIs enabled for fe.dai-link-1
[  830.245105][  T994]  fe.dai-link-1: ASoC: dpcm_fe_dai_prepare() failed (-22)
```
Is caused by this bad boy, because n2 doesn't have an SPDIF port and we don't route anything on PCM1 (`# AXG S/PDIF` in soundconfig)
Per se, it's not a problem since PA won't show this card.

The simpler would be to get rid of patches adding those conf files (only keep the Makefile.am change), and move the conf files into SRC_URI.

Then add a variant of the `axg-sound-card.conf` into the `files/hardkernel-odroidn2` subdirectory without the axg-sound-card.pcm.iec958.0 card.

We should probably do the reverse, but it's not a big deal.

Reply:
Done